pkg/services/ngalert/schedule/schedule.go

```diff
@@ -111,6 +111,12 @@ type SchedulerCfg struct {
 // NewScheduler returns a new schedule.
 func NewScheduler(cfg SchedulerCfg, dataService *tsdb.Service, appURL string, stateManager *state.Manager) *schedule {
 	ticker := alerting.NewTicker(cfg.C.Now(), time.Second*0, cfg.C, int64(cfg.BaseInterval.Seconds()))
+	appUrl, err := url.Parse(appURL)
```

Comment:
It makes sense to me to move the parsing of the URL to the code that calls `NewScheduler` and for `NewScheduler` to expect a `*net.URL` instead of `string`?

Reply:
Ideally, I think we should parse it in the code that creates the global config struct but I did not go that far because it would cause changes in many places.